src/klr/klr.cpp

```diff
@@ -373,5 +374,25 @@ int CallFirmwareProcessMain(int argc, wchar_t* argv[])
 
 int wmain(int argc, wchar_t* argv[])
 {
+    // Check for the debug flag before doing anything else
+    for (int i = 0; i < argc; i++)
+    {
+        if (StringsEqual(argv[i], L"--debug"))
+        {
+            if (!IsDebuggerPresent())
+            {
+                ::wprintf_s(L"Waiting for the debugger to attach...\r\n");
+
+                // Do not use getchar() like in corerun because it doesn't work well with remote sessions
+                while (!IsDebuggerPresent())
+                {
+                    Sleep(250);
+                }
```

Comment:
Should we have set a time limitation, when beyond the limit the program should keep running without debugger?